File: nodejs/src/factory/products/ItalianTea.ts
class ItalianTea {
    brew() {
        return "Pha trà kiểu Ý";
    }
    getPrice() {
        return 40000;
    }
    getDescription() {
        return "Trà Ý";
    }
    getCost() {
        return this.getPrice();
    }
}

export default ItalianTea;

File: spring/demo/src/main/java/com/example/demo/factory/products/ItalianTea.java
package com.example.demo.factory.products;

public class ItalianTea implements Tea {
    @Override public String brew() { return "Pha Tè alla pesca kiểu Ý"; }
    @Override public int getPrice() { return 40000; }
    @Override public String getDescription() { return "Tè alla pesca"; }
    @Override public int getCost() { return getPrice(); }
}
